src/test/java/com/ibm/stocator/fs/swift2d/ObjectStoreFileSystemTest.java

```diff
@@ -26,35 +33,160 @@ public final void before() {
   @Test
   public void getObjectNameTest() throws Exception {
     Path input = new Path("swift2d://out1003.lvm/a/b/c/m.data/_temporary/"
-        + "0/_temporary/attempt_201603141928_0000_m_000099_102/part-00099");
+            + "0/_temporary/attempt_201603141928_0000_m_000099_102/part-00099");
     String result = Whitebox.invokeMethod(mMockObjectStoreFileSystem, "getObjectName", input,
-        Constants.HADOOP_TEMPORARY, true);
+            Constants.HADOOP_TEMPORARY, true);
     Assert.assertEquals("/a/b/c/m.data/part-00099-attempt_201603141928_0000_m_000099_102", result);
 
     input = new Path("swift2d://out1003.lvm/a/b/m.data/_temporary/"
-        + "0/_temporary/attempt_201603141928_0000_m_000099_102/part-00099");
+            + "0/_temporary/attempt_201603141928_0000_m_000099_102/part-00099");
     result = Whitebox.invokeMethod(mMockObjectStoreFileSystem, "getObjectName", input,
-        Constants.HADOOP_TEMPORARY, true);
+            Constants.HADOOP_TEMPORARY, true);
     Assert.assertEquals("/a/b/m.data/part-00099-attempt_201603141928_0000_m_000099_102", result);
 
     input = new Path("swift2d://out1003.lvm/m.data/_temporary/"
-        + "0/_temporary/attempt_201603141928_0000_m_000099_102/part-00099");
+            + "0/_temporary/attempt_201603141928_0000_m_000099_102/part-00099");
     result = Whitebox.invokeMethod(mMockObjectStoreFileSystem, "getObjectName", input,
-        Constants.HADOOP_TEMPORARY, true);
+            Constants.HADOOP_TEMPORARY, true);
     Assert.assertEquals("/m.data/part-00099-attempt_201603141928_0000_m_000099_102", result);
 
   }
 
   @Test(expected = IOException.class)
   public void getObjectWrongNameTest() throws Exception {
     Path input = new Path("swift2d://out1003.lvm_temporary/"
-        + "0/_temporary/attempt_201603141928_0000_m_000099_102/part-00099");
+            + "0/_temporary/attempt_201603141928_0000_m_000099_102/part-00099");
     Whitebox.invokeMethod(mMockObjectStoreFileSystem, "getObjectName", input,
-        Constants.HADOOP_TEMPORARY, true);
+            Constants.HADOOP_TEMPORARY, true);
 
     input = new Path("swift2d://out1003.lvm/temporary/"
-        + "0/_temporary/attempt_201603141928_0000_m_000099_102/part-00099");
+            + "0/_temporary/attempt_201603141928_0000_m_000099_102/part-00099");
     Whitebox.invokeMethod(mMockObjectStoreFileSystem, "getObjectName", input,
-        Constants.HADOOP_TEMPORARY, true);
+            Constants.HADOOP_TEMPORARY, true);
+  }
+
+  @Test
+  public void getSchemeTest() throws Exception {
+    Assume.assumeNotNull(getFs());
```

Comment:
Yea I was thinking about that, but there are test cases that do not require getFs() to be available. For example, getObjectNameTest() is written completely using Power Mock so it doesn't need to getFs(). If I put assumeNotNull(getFs()) in setUp(), then getObjectNameTest() would not run if FileSystem is not present. Not sure if I want to change the behavior.